arras-energy/gridlabd-old
# Enable httpd and mysqld to start from within the container.

#### What's this Pull Request do?
Ensures httpd and mysqld can successfully be started from within the container. 

#### Where should the reviewer start?
Review dockerfile and startup.sh script changes

#### How should this be tested?

`docker build -t supriyap/gridlabd:latest -f DockerFile .`

`docker run -v$(pwd)/data:/var/lib/mysql --rm -dit -p 80:80 supriyap/gridlabd:latest`

Navigate to localhost on the browser to see the default httpd startup page.
Also do a docker top <container id> to verify that mysqld is running. You can alternatively verify mysql by execing into the container to.

Eg:
     `docker exec -it <container id> bash`
     `mysql`

#### Any background context you want to provide?
Services are started using systemd from within centos containers. Using systemd inside of containers is troublesome due to D-BUS permission issues. In order to use systemd, we need to spin up privileged containers with sys/fs/cgroups mounted into the container. Usually containers must be started with the least privilege that are required. By skipping the reliance on systemd to start services, we can still have a non privileged container.

#### What are the relevant issues?
#### Screenshots (if appropriate)
#### Questions:
